packages/app/src/app/pages/SignIn/components/UserNameSelection.tsx

```diff
@@ -1,88 +1,149 @@
+];
+
+const usageOptions = [
+  { value: 'freelance-or-client', label: 'Freelance or client work' },
+  { value: 'startup', label: 'Startup' },
+  { value: 'enterprise', label: 'Enterprise' },
+  { value: 'hobby', label: 'Hobby or fun' },
+  { value: 'education', label: 'Education' },
+  { value: 'other', label: 'Other' },
+];
 
 export const UserNameSelection = () => {
-  const { pendingUser } = useAppState();
+  /**
+   * 🚧 Utility to debug Trial Onboarding Questions
+   */
+  const TOQ_DEBUG = window.localStorage.getItem('TOQ_DEBUG') === 'ENABLED';
+
+  // 🚧 Remove
+  let pendingUser = useAppState().pendingUser;
+
+  // 🚧 Uncomment
+  // const { pendingUser } = useAppState();
+
+  // 🚧 Remove
+  if (TOQ_DEBUG) {
+    pendingUser = {
+      avatarUrl: 'https://avatars.githubusercontent.com/u/7533849?v=4',
+      username: 'tristandubbeld',
+      name: 'Tristan Dubbeld',
+      id: 'id',
+      valid: true,
+    };
+  }
+
   const { validateUsername, finalizeSignUp } = useActions();
-  const [newUsername, setNewUsername] = useState('');
+  const [newUsername, setNewUsername] = useState(pendingUser?.username || '');
+  const [newDisplayName, setNewDisplayName] = useState(pendingUser?.name || '');
   const [loadingUsername, setLoadingUserName] = useState(false);
-  const [newDisplayName, setNewDisplayName] = useState('');
+  const firstName = pendingUser?.name.split(' ')[0];
 
-  useEffect(() => {
-    setNewUsername(pendingUser?.username);
-  }, [pendingUser]);
+  return (
+    <Stack
+      direction="vertical"
+      gap={4}
+      css={{ width: '100%', maxWidth: '370px' }}
+    >
+      <Stack direction="vertical" gap={6} align="center">
+        <Element
+          as="img"
+          css={{
+            width: 64,
+            height: 64,
+            borderRadius: '100%',
+          }}
+          alt={pendingUser.username}
+          src={pendingUser.avatarUrl}
+        />
+        <Text
+          as="h1"
+          size={32}
+          weight="500"
+          align="center"
+          css={{
+            margin: 0,
+            color: '#ffffff',
+            fontFamily: 'Everett, sans-serif',
+            lineHeight: '42px',
+            letterSpacing: '-0.01em',
+          }}
+        >
+          Welcome{firstName ? `, ${firstName}` : null}
+        </Text>
+      </Stack>
+      <Stack
+        as="form"
+        direction="vertical"
+        gap={6}
+        onSubmit={e => {
+          e.preventDefault();
+          finalizeSignUp({ username: newUsername, name: newDisplayName });
+        }}
+      >
+        <Stack direction="vertical" gap={4}>
+          <InputText
+            id="username"
+            name="username"
+            label="Username"
+            // TODO: Instead of onBlur we can debounce onchange
+            onBlur={async e => {
+              setLoadingUserName(true);
+              await validateUsername(e.target.value);
+              setLoadingUserName(false);
+            }}
```

Comment:
Is this working? I wasn't able to verify it.

Reply:
It should work exactly the same as it did before the "facelift". I will see if I can create a way to properly verify it in some way. Thanks.